packages/openid4vp/src/openid4vp-auth-request/verify-openid4vp-auth-request.ts

```diff
@@ -0,0 +1,80 @@
+import type { CallbackContext } from '@openid4vc/oauth2'
+import * as v from 'valibot'
+import { parseClientIdentifier } from '../client-identifier-scheme/parse-client-identifier-scheme'
+import { verifyJarRequest } from '../jar/handle-jar-request/verify-jar-request'
+import { type JarAuthRequest, vJarAuthRequest } from '../jar/v-jar-auth-request'
+import type { WalletMetadata } from '../models/v-wallet-metadata'
+import { parseTransactionData } from '../transaction-data/parse-transaction-data'
+import { type Openid4vpAuthRequest, vOpenid4vpAuthRequest } from './v-openid4vp-auth-request'
+import { validateOpenid4vpAuthRequestParams } from './validate-openid4vp-auth-request'
+
+export async function verifyOpenid4vpAuthRequest(
+  params: Openid4vpAuthRequest | JarAuthRequest,
+  options: {
```

Comment:
Genereally would like to see more interfaces (in oid4vci most methods have a separate options interface)